kernel/syscall.c

```diff
@@ -3126,6 +3126,57 @@ long myst_syscall_mbind(
     return ret;
 }
 
+long myst_syscall_get_mempolicy(
+    int* mode,
+    unsigned long* nodemask,
+    unsigned long maxnode,
+    void* addr,
+    unsigned long flags)
+{
+    unsigned long rounded_maxnode = 0;
+    int ROUND_UP_VAL = sizeof(unsigned long) * 8;
+    myst_round_up(maxnode, ROUND_UP_VAL, &rounded_maxnode);
+
+    unsigned long* rounded_nodemask;
+    // allocate higher buffer
+    if (rounded_maxnode > maxnode)
```

Comment:
`nodemask` buffer size is supposed to be big enough to hold `maxnode` of bits. The API does not provide any parameter to specify the size of the `nodemask` buffer. Therefore the Linux manual clarified that the implementation of API can round up `maxnode` to the next multiple of `sizeof(unsigned long) * 8` (bits in one element of the `nodemask` buffer), and use it to derive the buffer size. No need to allocate an internal buffer if `maxnode` provided is not a multiple of `sizeof(unsigned long) * 8`

Reply:
@mingweishih and I actually tried this case out, and got a segmentation fault in the case where we pass a non multiple, since we are rounding the maxnode value before passing it to the ocall. If we are passing it a higher value, it seems to also be expecting a larger buffer. My guess is if we remove the rounding we also remove the need for a larger buffer, since it would all be taken care of by the ocall/linux call itself